src/API/Google/ExceptionTrait.php

```diff
@@ -44,6 +46,36 @@ protected function has_api_exception_error( ApiException $exception, string $err
 		return false;
 	}
 
+	/**
+	 * Returns a list of detailed errors from an exception instance that extends ApiException
+	 * or GoogleServiceException. Other Exception instances will also be converted to an array
+	 * in the same structure.
+	 *
+	 * The following are the example sources of ApiException, GoogleServiceException,
+	 * and other Exception in order:
+	 *
+	 * @link https://github.com/googleads/google-ads-php/blob/v19.1.0/src/Google/Ads/GoogleAds/V13/Services/Gapic/CustomerServiceGapicClient.php#L359-L361
+	 * @link https://github.com/googleapis/google-api-php-client/blob/v2.15.0/src/Http/REST.php#L119-L135
+	 * @link https://github.com/googleapis/google-api-php-client/blob/v2.15.0/src/Service/Resource.php#L86-L175
+	 *
+	 * @param ApiException|GoogleServiceException|Exception $exception Exception to check.
+	 *
+	 * @return array
+	 */
+	protected function get_exception_errors( Exception $exception ): array {
```

Comment:
In the docs it says --> `ApiException|GoogleServiceException|Exception `

However, in the type only Exception is allowed.

Reply:
Support for union type declarations was added in PHP 8.0 but GLA needs to be compatible with PHP >= 7.4. Considering both `ApiException` and `GoogleServiceException` extend the `Exception` class, so I thought it should be okay to declare `Exception` here and be a bit better than no type declaration.

However, I'm not that familiar with PHP, does this sound reasonable?

Reply:
Yes, I didn't get into that compatibility problem. 

What do you think about removing the type in the param, tho? 

`protected function get_exception_errors( $exception ): array `

Reply:
> Yes, I didn't get into that compatibility problem.
> 
> What do you think about removing the type in the param, tho?
> 
> `protected function get_exception_errors( $exception ): array `

If I understood it right, `mixed` is assumed when no type is declared, which means it would be broader than expected types. Moreover, most PHP files have `declare( strict_types=1 );` in this codebase, I assumed the point of enabling strict types is also to catch type errors earlier. So I considered it better to declare type when possible.

Do you think it's better not to declare the type in this use case?

Reply:
> it better to declare type when possible.

Yes, I agree with this, the issue for me was that I thought Exception was another Google-based Exception alias excluding ApiException and GoogleServiceException but I checked again and seems to be a generic PHP Exception. So you're right and the Exception type is the most correct. I wonder if we should add just Exception as a type in the docs since we are already handling the interfaces ( via $exception instanceof  ) and ApiException and GoogleServiceException inherit from Exception. BUt this is not anymore a blocker for me.

BTW, Strict typing is only defined for scalar type declarations. So it doesn't apply here. ([More info](https://www.php.net/manual/en/language.types.declarations.php#language.types.declarations.strict)) 

In regards to `mixed`, you're right. It's equivalent to "object|resource|array|string|float|int|bool|null"

Reply:
> BTW, Strict typing is only defined for scalar type declarations. So it doesn't apply here. ([More info](https://www.php.net/manual/en/language.types.declarations.php#language.types.declarations.strict))

Thanks for the correction!